### PR TITLE
adding in function for pulling nwis data 

### PR DESCRIPTION
**Pull request by jzwart**

### CODE REVIEW

File: 1_data/src/pull_nwis.R [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+pull_nwis <- function(ind_file, sites_yml, dates_yml, params_yml, gd_config) {
+
+  dates <- yaml::yaml.load_file(dates_yml)
+  pcodes <- yaml::yaml.load_file(params_yml)
+  site_ids <- yaml::yaml.load_file(sites_yml)
+
+  flow <- readNWISuv(siteNumbers=site_ids, parameterCd=pcodes$flow, startDate=dates$calibrate$start, endDate=dates$forecast$start)
+  nitrate <- readNWISuv(siteNumbers=site_ids, parameterCd=pcodes$nitrate, startDate=dates$calibrate$start, endDate=dates$forecast$start)
+
+  writeLines(list(flow=flow, nitrate=nitrate), outfile)
```
Comment (aappling-usgs) [inline 1_data/src/pull_nwis.R]: let's delete this one b/c it's duplicated below (and uses `outfile` which no longer exists in thix context)

File: remake.yml [hunk 2]
```diff
@@ -35,6 +35,8 @@ targets:
     command: pull_nwis(
       ind_file=target_name,
       sites_yml='1_data/out/sites.yml',
+      dates_yml='1_data/cfg/dates.yml',
+      params_yml='1_data/cfg/params.yml',
```
Comment (aappling-usgs) [inline remake.yml]: looks about right :+1:

File: 1_data/src/pull_nwis.R (hunk 1 above)
Comment (jzwart) [reply 1_data/src/pull_nwis.R]: oops good catch

File: 1_data/src/pull_nwis.R [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+  data_file <- as_data_file(ind_file)
+  writeLines(list(flow = flow, nitrate = nitrate), data_file)
```
[the remark this answers is not in the document]
Comment (jzwart) [reply 1_data/src/pull_nwis.R]: yup

**Pull request merged**